tests/model/test_model.py

```diff
@@ -1416,11 +1416,15 @@ def reaction_event_index_factory(self):
         Generate index for reaction tests based on minimal specification
 
         Input is a list of pairs, of a message-id and a list of reaction tuples
+        NOTE: reactions as None indicate not indexed, [] indicates no reaction
         """
-        def _factory(msgs: List[Tuple[int, List[Tuple[int, str, str, str]]]]):
+        MsgsType = List[Tuple[int, Optional[List[Tuple[int, str, str, str]]]]]
+
+        def _factory(msgs: MsgsType):
             return {
                 'messages': {
-                    message_id: {
+                    message_id:
+                    {} if reactions is None else {
```

Comment:
This is a pretty neat logic for simplifying generating both indexed/non-indexed reactions. :+1: